File: Pohon.php
<?php

// Kelas dasar
class Pohon {
    protected $tumbuh;
    protected $tinggi;
    protected $fotosintesis;
    protected $umur;

    // Constructor
    public function __construct($tumbuh, $tinggi, $fotosintesis, $umur) {
        $this->tumbuh = $tumbuh;
        $this->tinggi = $tinggi;
        $this->fotosintesis = $fotosintesis;
        $this->umur = $umur;
    }

    public function getTumbuh() {
        return $this->tumbuh;
    }

    public function getTinggi() {
        return $this->tinggi;
    }

    public function getFotosintesis() {
        return $this->fotosintesis;
    }

    public function getUmur() {
        return $this->umur;
    }

    public function cekUmur() {
        return 0.0;
    }

    // Destructor
    public function __destruct() {
        echo "Tampilkan objek Pohon<br>";
    }
}

// Kelas turunan
class PohonPisang extends Pohon {
    // Constructor
    public function __construct($tumbuh, $tinggi, $fotosintesis, $umur) {
        parent::__construct($tumbuh, $tinggi, $fotosintesis, $umur);
    }

    // Destructor
    public function __destruct() {
        echo "Tampilkan objek Pohon Pisang<br>";
    }
}

// Kelas turunan lainnya
class PohonMangga extends Pohon {
    // Constructor
    public function __construct($tumbuh, $tinggi, $fotosintesis, $umur) {
        parent::__construct($tumbuh, $tinggi, $fotosintesis, $umur);
    }

    // Destructor
    public function __destruct() {
        echo "Tampilkan objek Pohon Mangga<br>";
    }
}

// Contoh overloading
class ContohOverloading {
    public function cetak($num) {
        echo "Umur: $num<br>";
    }
}

// Contoh polimorfisme
function cetakUmur($pohon) {
    echo "Umur: " . $pohon->cekUmur() . "<br>";
}

$pohonPisang = new PohonPisang("Ke atas", 12000, "Siang hari", 1);
echo "Tumbuh: " . $pohonPisang->getTumbuh() . "<br>";
echo "Fotosintesis: " . $pohonPisang->getFotosintesis() . "<br>";
echo "Umur: " . $pohonPisang->getUmur() . "<br>";
echo "Tinggi: " . $pohonPisang->getTinggi() . "<br>";
cetakUmur($pohonPisang);

$pohonMangga = new PohonMangga("Kesamping", 20000, "Sore hari", 7);
echo "Tumbuh: " . $pohonMangga->getTumbuh() . "<br>";
echo "Fotosintesis: " . $pohonMangga->getFotosintesis() . "<br>";
echo "Umur: " . $pohonMangga->getUmur() . "<br>";
echo "Tinggi: " . $pohonMangga->getTinggi() . "<br>";
cetakUmur($pohonMangga);

$contoh = new ContohOverloading();
$contoh->cetak(1);
$contoh->cetak(2);

?>

File: Pohon.cpp
#include <iostream>
using namespace std;

// Kelas dasar
class Pohon {
protected:
    string tumbuh;
    int tinggi;
    string fotosintesis;
    int umur;

public:
    // Constructor
    Pohon(string tumbuh, int tinggi, string fotosintesis, int umur) : tumbuh(tumbuh), tinggi(tinggi), fotosintesis(fotosintesis), umur(umur) {}

    string gettumbuh() {
        return tumbuh;
    }

    int gettinggi() {
        return tinggi;
    }
    string getfotosintesis() {
        return fotosintesis;
    }
    int getumur() {
        return umur;
    }

    virtual double cekUmur() {
        return 0.0;
    }

    // Destructor
    virtual ~Pohon() {
        cout << "Tampilkan objek Pohon" << endl;
    }
};

// Kelas turunan
class Pohonpisang : public Pohon {
public:
    // Constructor
    Pohonpisang(string tumbuh, int tinggi, string fotosintesis, int umur) : Pohon(tumbuh, tinggi, fotosintesis, umur) {}

    string gettumbuh() {
        return tumbuh;
    }

    int gettinggi() {
        return tinggi;
    }
    string getfotosintesis() {
        return fotosintesis;
    }
    int getumur() {
        return umur;
    }

    // Destructor
    ~Pohonpisang() {
        cout << "Tampilkan objek Pohonpisang" << endl;
    }
};

// Kelas turunan lainnya
class Pohonmangga : public Pohon {
public:
    // Constructor
    Pohonmangga(string tumbuh, int tinggi, string fotosintesis, int umur) : Pohon(tumbuh, tinggi, fotosintesis, umur) {}

    string gettumbuh() {
        return tumbuh;
    }

    int gettinggi() {
        return tinggi;
    }
    string getfotosintesis() {
        return fotosintesis;
    }
    int getumur() {
        return umur;
    }

    // Destructor
    ~Pohonmangga() {
        cout << "Tampilkan objek Pohonmangga" << endl;
    }
};

// Contoh overloading
class ContohOverloading {
public:
    void cetak(int num) {
        cout << "Umur: " << num << endl;
    }

    void cetak(double num) {
        cout << "Umur: " << num << endl;
    }
};

// Contoh polimorfisme
void cetakUmur(Pohon* pohon) {
    cout << "Umur: " << pohon->cekUmur() << endl;
}

int main() {
    Pohonpisang pohonpisang("Ke atas", 12000, "Siang hari", 1);
    cout << "Tumbuh: " << pohonpisang.gettumbuh() << endl;
    cout << "Fotosintesis: " << pohonpisang.getfotosintesis() << endl;
    cout << "Umur: " << pohonpisang.getumur() << endl;
    cout << "Tinggi: " << pohonpisang.gettinggi() << endl;
    cetakUmur(&pohonpisang);

    Pohonmangga pohonmangga("Kesamping", 20000, "Sore hari", 7);
    cout << "Tumbuh: " << pohonmangga.gettumbuh() << endl;
    cout << "Fotosintesis: " << pohonmangga.getfotosintesis() << endl;
    cout << "Umur: " << pohonmangga.getumur() << endl;
    cout << "Tinggi: " << pohonmangga.gettinggi() << endl;
    cetakUmur(&pohonmangga);

    ContohOverloading contoh;
    contoh.cetak(1);
    contoh.cetak(2);

    return 0;
}